### ISSUE TITLE
ReflectionExtensions should not be public

### ISSUE THREAD

**Issue by j2jensen** [RicoSuter/NJsonSchema#699]
I keep getting intellisense suggestions on random objects for the extension methods in `NJsonSchema.Infrastructure.ReflectionExtensions`. Since these extensions are not intended to be a part of your API, they should be `internal`.

**Comment by RSuter** [comment]
Good point, the problem is that NSwag which is built on top of NJS is using these APIs too... would just hiding them from intellisense help?

**Comment by RSuter** [comment]
But yes, we need a better solution... maybe InternalsVisibleTo or another lib.
